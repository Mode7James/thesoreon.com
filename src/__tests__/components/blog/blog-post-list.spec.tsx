import React from "react"
import { render } from "@testing-library/react"

import BlogPostList from "~/components/blog/blog-post-list"
import ThemeProvider from "~/components/theme-provider/theme-provider"

describe("BlogPostList component", () => {
  const renderComponent = () => {
    return render(
      <ThemeProvider>
        <BlogPostList
          data={{
            allMdx: {
              edges: [
                {
                  node: {
                    fields: {
                      slug: "/hello-world",
                    },
                    frontmatter: {
                      title: "Hello World!",
                      tags: ["test", "test2"],
                      date: "11-13-2000",
                    },
                  },
                },
              ],
            },
          }}
        />
      </ThemeProvider>
    )
  }

  test("should render", () => {
    const { container } = renderComponent()

    expect(container).toMatchSnapshot()
  })
})
