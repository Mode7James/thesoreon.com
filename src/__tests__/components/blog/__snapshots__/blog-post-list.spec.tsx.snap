// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPostList component should render 1`] = `
.emotion-1 {
  margin: 8px 0;
}

.emotion-0 {
  color: var(--links-color);
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: 100%;
  margin: 4px 0;
  outline: none;
}

.emotion-0 h2 {
  display: inline;
}

.emotion-0:active:active::before {
  content: "";
}

.emotion-0:focus {
  color: var(--links-focus-color);
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:focus::before {
  content: "â–¶ ";
  vertical-align: text-top;
}

<div>
  <div
    class="emotion-1"
  >
    <a
      class="emotion-0"
      href="/hello-world"
    >
      <h2>
        Hello World!
      </h2>
    </a>
  </div>
</div>
`;
